### PR TITLE
fix(server): devCDN should not crash when bundles are not present

### PR DESCRIPTION
**Pull request by PixnBits**
## Description

The development CDN crashes when performing the substitution in the URL of a nonexistant bundle.

## Motivation and Context

The legacy browser bundle is optional, not required.

```
err: {
    "type": "TypeError",
    "message": "Cannot read properties of undefined (reading 'url')",
    "stack":
        TypeError: Cannot read properties of undefined (reading 'url')
            at .../one-app/lib/server/utils/devCdnFactory.js:44:53
```

## How Has This Been Tested?

Unit tests added that have only one bundle each.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Resolve the server crashing on them when running with modules that have not opted in to building the legacy browser bundles.

### CODE REVIEW

File: src/server/utils/devCdnFactory.js [hunk 1]
```diff
@@ -33,9 +33,15 @@ const getLocalModuleMap = ({ pathToModuleMap, oneAppDevCdnAddress }) => {
   const moduleMap = JSON.parse(fs.readFileSync(pathToModuleMap, 'utf8').toString());
   Object.keys(moduleMap.modules).forEach((moduleName) => {
     const module = moduleMap.modules[moduleName];
-    module.browser.url = module.browser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
-    module.legacyBrowser.url = module.legacyBrowser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
-    module.node.url = module.node.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    Object
+      .values(module)
+      .filter((bundle) => Object.hasOwnProperty.call(bundle, 'url'))
```
Comment (PixnBits) [inline src/server/utils/devCdnFactory.js:38]: this is defensive; do we need to be? e.g. is `baseUrl` always computed or can it be part of the module map JSON resource?

**Review by code-forger — APPROVED** [review]

**Review by JAdshead — DISMISSED** [review]

**Review by code-forger — CHANGES_REQUESTED** [review]
This Pr fixes a symptom, not a cause, the root cause is fixed elsewhere

File: src/server/utils/devCdnFactory.js [hunk 2]
```diff
@@ -72,15 +78,15 @@ const consumeRemoteRequest = async (remoteModuleMapUrl, hostAddress, remoteModul
...
-      );
+      Object
+        .values(module)
+        .filter((bundle) => Object.hasOwnProperty.call(bundle, 'url'))
+        .forEach((bundle) => {
+          /* eslint-disable-next-line no-param-reassign -- the in-memory copy is created here during
+          // the read from network, so the replacement side-effect of this loop is local to the
+          // consumeRemoteRequest function and not to any arguments */
+          bundle.url = bundle.url.replace(new URL(bundle.url).origin, oneAppDevStaticsAddress);
+        });
```
Comment (PixnBits) [inline src/server/utils/devCdnFactory.js:89]: I'm not sure where the test(s) for the previous implementation is/are, so this needs attention

**Comment by PixnBits** [comment]
> This Pr fixes a symptom, not a cause, the root cause is fixed elsewhere

https://github.com/americanexpress/one-app-cli/pull/629

**Pull request closed without merging**